### PR TITLE
Add switch field to nodes API

### PR DESCRIPTION
**Pull request by mpolden**
This changes node allocation to prefer nodes on network switches distinct within
a cluster. We do this to avoid allocating multiple nodes in the same cluster on
the same top-of-rack switch, thereby reducing the impact of an entire rack
dying.

Behaviour:
- Deployment prefers nodes on exclusive switches.
- Deployment reuses switches instead of failing if exclusive allocation isn't
  possible.
- Node lacking switch information are exclusive by default to avoid unnecessary
  reallocation. This is currently the case for all nodes as none have switch
  information yet.

Merge together with internal PR.

@bratseth

### CODE REVIEW

**Review by bratseth — COMMENTED** [review]
Looks great, just one question.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeCandidate.java [hunk 1]
```diff
@@ -57,14 +62,18 @@
...
-     * @return negative if first priority is higher than second node
+     * @return negative if this should be preferred over other
      */
     @Override
     public int compareTo(NodeCandidate other) {
         // First always pick nodes without violation above nodes with violations
         if (!this.violatesSpares && other.violatesSpares) return -1;
         if (!other.violatesSpares && this.violatesSpares) return 1;
 
+        // Prefer node on exclusive switch
```
[the remark this answers is not in the document]
Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeCandidate.java:73]: It should? That means we'll never prefer a new node on a exclusive switch over an existing active node.

[the remark this answers is not in the document]
Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeCandidate.java:73]: The way it is now we'll always (if this code gets its way) prefer new nodes to existing ones if the switch distribution is not optimal. I don't think we want that but instead want to do a controlled retire attempt node by node initiated by a maintainer?

[the remark this answers is not in the document]
Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeCandidate.java:73]: To spread out retiring in time? OK, I buy that.

**Comment by mpolden** [comment]
Dropped the allocation changes in this PR. Only the new field and REST API changes remain.

**Review by bratseth — APPROVED** [review]

**Pull request merged**